src/software/simulated_tests/corner_kick_play_test.cpp

```diff
@@ -29,19 +34,10 @@ TEST_F(CornerKickPlayTest, test_corner_kick_play)
     setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::INDIRECT_FREE_US);
 
     std::vector<ValidationFunction> terminating_validation_functions = {
-        // The ball must enter the enemy net in order for the test to pass. This is to
-        // temporarily
-        // prevent regressions like
-        // https://github.com/UBC-Thunderbots/Software/issues/1690
-        // until we have proper validation functions built up.
-        // TODO: Implement proper validation
-        // https://github.com/UBC-Thunderbots/Software/issues/1396
         [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-            while (
-                !contains(world_ptr->field().enemyGoal(), world_ptr->ball().position()))
-            {
-                yield();
-            }
+            robotReceivedBall(world_ptr->friendlyTeam().getAllRobots()[0], world_ptr, yield);
```

Comment:
Perhaps leave a TODO here linked to a ticket to also add a check for the robot kicking the ball? IIRC we were doing that later after some changes were made to how we detect kicks since it was non-blocking for now.

Reply:
added a question to this PR about this